compiler-rt/lib/profile/InstrProfilingWriter.c

```diff
@@ -214,6 +221,37 @@ static int writeOneValueProfData(ProfBufferIO *BufferIO,
   return 0;
 }
 
+/* Compute size of VPData, without actually writing it. This function follows
+ * the logic of writeValueProfData below it. It is used for predicting how big
+ * a buffer is needed to store the ValueProfileData.
```

Comment:
nit: If possible, introduce a common helper function and call it from both `__llvm_profile_getSizeOfValueProfData` and `writeValueProfData` to de-duplicate logic.